docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+----------------------------------------------------------
+
+`--validation-dataset` (string) (*Release 0.1*)
+
+File path pointing to the validation dataset file. The validation dataset is optional, in any case.
+
+If using a `validation dataset`, this is how the behaviour should be:
+
+- The `test-dataset` and `--dataset` arguments are also required.
+
+- The `validation-dataset` dataset is used to estimate prediction error for model selection. 
+
+- The `test-dataset` is used for assessment of the generalization error of the final chosen model. Ideally, the test set should be kept in a “vault,” and be brought out only at the end of the data analysis.
+
+Basically, when using a `validation dataset` plus the `test dataset`, the validation phase is split into two parts:
+
+1. In the first part you just look at your models and select the best performing approach using the validation data (=validation)
+2. Then you estimate the accuracy of the selected approach (=test).
+	
+Hence, the separation of data could be 80/10/10 or 75/15/10. For example:
+
+- `training-dataset` file should have 75% of the data.
+- `validation-dataset` file should have 15% of the data.
+- `test-dataset` file should have 10% of the data.
+
+In any case, those percentages will be decided by the user using the CLI who will provide the files already split.
+
+----------------------------------------------------------
+
+`--label-column-name` (string) (*Release 0.1*)
+
+With this argument, a specifc objective/target column (the variable that you want to predict) can be specified by using the column's name set in the dataset's header. 
+
+If you do not explicitly specify a label column, the CLI will automatically infer what's the label column in the dataset (To be implemented in v0.2 or v1.0).
+
+This argument is used only for supervised ML task such as a *classification problem*. It cannot be used for unsupervised ML Tasks such as *clustering*.
+
+----------------------------------------------------------
+
+`--label-column-index` (int) (*Release 0.1*)
+
+With this argument, a specifc objective/target column (the variable that you want to predict) can be specified by using the column's numeric index in the dataset's file (Starting on 1. First column would be 1).
+
+*Note:* If the user is also using the `--label-column-name`, the `--label-column-name` is the one being used.
+
+If you do not explicitly specify a label column, the CLI will to automatically infer what's the label column in the dataset (To be implemented in v0.2 or v1.0).
+
+This argument is used only for supervised ML task such as a *classification problem*. It cannot be used for unsupervised ML Tasks such as *clustering*.
+
+----------------------------------------------------------
+
+`--has-header` (bool) (*Release 0.1*)
+
+Specify if the dataset file(s) have a header row.
+Possible values are: 
+- `true`
+- `false`
+
+The by default value is `true`. 
```

Comment:
Does this mean that if I specify `--has-header` on the command line, it will default to `true`? Or does it mean that if I DON'T specifiy `--has-header` it will default to `true`?

Reply:
I think in either case it defaults to true.

Reply:
If so, then the parameter name should be flipped the other way `--no-header`. It doesn't make sense to have a command:

```
foo.exe
```

and 

```
foo.exe --my-option
```

Do the same things.

Reply:
Should this rule be applied to all Boolean arguments with default values?

Reply:
Typically, yes.

Look at any other command line. Let's say `git`.

```
git-fetch - Download objects and refs from another repository

SYNOPSIS
git fetch [<options>] [<repository> [<refspec>…​]]
git fetch [<options>] <group>
git fetch --multiple [<options>] [(<repository> | <group>)…​]
git fetch --all [<options>]
```

By default, `git fetch` won't fetch from all remotes. But when you specify `git fetch --all`, it does.

Reply:
@CESARDELATORRE FYI